crates/integration-tests/dns-ptr.lua

```diff
@@ -0,0 +1,28 @@
+local kumo = require 'kumo'
+
+local function contains(list, t)
+    for key, value in pairs(list) do
+        if value == t then
+            return true
+        end
+    end
+    return false
+end
+
+-- check if we're able to resolve back to localhost
+local ok, a = pcall(kumo.dns.lookup_ptr, '127.0.0.1')
+assert(ok, 'expected localhost for 127.0.0.1 ptr')
+if ok then
+  assert(contains(a, 'localhost.'), 'expected localhost.')
+end
+
+-- see if we're able to do resolve for ipv6
+local ok, a = pcall(kumo.dns.lookup_ptr, '::1')
+assert(ok, 'expected localhost for ::1 ptr')
+if ok then
+  assert(contains(a, 'localhost.'), 'expected localhost.')
+end
+
+local ok, a = pcall(kumo.dns.lookup_ptr, '::2')
+assert(not ok, '::2 should not have ptr')
```

Comment:
With the test resolver configuration suggested above, this PTR lookup returns an empty list of records rather than an error, just because the test resolver prefers to return empty results rather than an error in that case.

I don't think it is strictly necessary to have this test case here, but it is something that you might want to consider when you build your larger integration as it depends on the zone being queried and the DNS server configuration whether you will experience an error or an empty result set.